scripts/install.sh

```diff
@@ -221,7 +221,8 @@ check_http_status() {
 is_dir_in_path() {
   dir="$1"
   found=1
-  (echo "$PATH" | grep -q "$dir:" || echo "$PATH" | grep -q -E "$dir$") || found=0
+  # ensure dir is the full path and not a substring of some longer path
+  ( (echo "$PATH" | grep -q -E "(^|:)$dir(:|$)") || (echo "$PATH" | grep -q -E "$dir$") ) || found=0
```

Comment:
It looks like we're using regex here already but are there any concerns about having special characters in the path? `[]` for example

Reply:
I believe this would already handle special characters like that. Am I overlooking something glaring?

Reply:
The one I'm mainly worried about is `.` -- For example, `foo.biz` would match `foolbiz` or `foo/biz`

Reply:
Special characters like `.` could indeed result in an erroneous match. Practically, we don't need to worry about this today because the `is_dir_in_path` check is only performed against a static list of paths (`/usr/local/bin`, `/usr/bin`, `/usr/sbin`). I'll add a comment to the function noting this.

Reply:
I ended up being able to clean this up a bit by performing a fixed-string (i.e. non-regex) search against the regex search results. This should prevent most special characters from getting in the way, unless they cause the initial regex search to miss results.

https://github.com/DopplerHQ/cli/compare/7d25f66544f587f77c58af09535f82dad222f564..475352a2fb77679e94e906048f0e5ee78fc5dc25